### PR TITLE
UPDATE python-jose to 3.5.0 and enhance JWT validation

### PR DESCRIPTION
**Pull request by ramesh-maddegoda**
## 🗒️ Summary

This PR upgrades the `python-jose` library to version 3.5.0 and enhances JWT validation in the Nucleus Cognito Lambda authentication function.

**Changes:**
- Upgraded `python-jose[cryptography]` to `~=3.5.0` in Lambda requirements
- Added explicit algorithm validation for JWT tokens (ES256 for ALB OIDC, RS256 for Cognito tokens)
- Enhanced JWT validation to verify algorithm header before decoding
- Fixed Lambda build script Docker permission issues using `--user` flag
- Updated Terraform provisioner to use explicit bash interpreter
- Updated RDS cluster engine version default to `8.0.mysql_aurora.3.10.3`

**Benefits:**
- Improved JWT validation robustness
- Better alignment with security best practices
- Cleaner Lambda build process

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90 %

## ⚙️ Test Data and/or Report

**Manual Testing:**
1. Lambda build script successfully creates deployment package with upgraded dependencies
2. Docker container runs with proper user permissions (no permission denied errors)
3. Terraform provisioner executes build script without shell interpreter issues
4. Latest deployed code in lambda to reflects the changes (E.g.: allowed algorithms)
5. Successfully tested authentication with Cognito login to Nucleus
6. Confirmed RDS cluster remains at version `8.0.mysql_aurora.3.10.3`

## ♻️ Related Issues

Fixes https://github.com/NASA-PDS/outlaw-tracker/issues/35

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [x] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [x] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### CODE REVIEW

File: terraform/terraform-modules/cognito-auth/lambda/build-lambda.sh [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
+echo "[$(date '+%Y-%m-%d %H:%M:%S')] Starting Lambda package build..."
 
-# Move to script directory
+# Navigate to the script's directory regardless of where it was invoked
 cd "$(dirname "$0")"
 
-# Clean old build and create package folder
-rm -rf package && mkdir -p package
+# Wipe the old directory cleanly
+rm -rf package
```
Comment (nutjob4life) [inline terraform/terraform-modules/cognito-auth/lambda/build-lambda.sh]: Should we also wipe out any old `lambda_package.zip` from previous runs?

File: terraform/terraform-modules/cognito-auth/cognito-auth.tf [hunk 2]
```diff
@@ -48,28 +48,16 @@ data "aws_region" "current" {}
...
   }
 
-  triggers = {
-    dependencies_versions = filemd5("${path.module}/lambda/requirements.txt")
-    source_versions       = filemd5("${path.module}/lambda/pds_nucleus_alb_auth.py")
-    build_script_version  = filemd5("${path.module}/lambda/build-lambda.sh")
+  provisioner "local-exec" {
+    # Using explicit interpreter array prevents /bin/sh fallback inconsistencies
+    interpreter = ["/bin/bash", "-c"]
+    command     = "cd ${path.module}/lambda && ./build-lambda.sh"
```
Comment (nutjob4life) [inline terraform/terraform-modules/cognito-auth/cognito-auth.tf]: If `path.module` contains spaces, this `command` could break on the `cd` — probably won't happen but sometimes it's fun to be thorough 😉

**Review by nutjob4life — APPROVED** [review]
Delta: ✓ (based on visual inspection only; trust accordingly)
Tests: not run
Approval: ✅
Comment: I did leave 2 comments on 2 separate lines in the diffs; feel free to read, respond, or even ignore as I don't think they're critical

File: terraform/terraform-modules/cognito-auth/lambda/build-lambda.sh (hunk 1 above)
Comment (ramesh-maddegoda) [reply terraform/terraform-modules/cognito-auth/lambda/build-lambda.sh]: Yes, I updated it with https://github.com/NASA-PDS/nucleus/pull/204/changes/cb3ae53816df26daaec520cbeac231c97377bfdd

File: terraform/terraform-modules/cognito-auth/cognito-auth.tf (hunk 2 above)
Comment (ramesh-maddegoda) [reply terraform/terraform-modules/cognito-auth/cognito-auth.tf]: Good catch! Fixed it with https://github.com/NASA-PDS/nucleus/pull/204/changes/cb3ae53816df26daaec520cbeac231c97377bfdd

**Comment by ramesh-maddegoda** [comment]
> Delta: ✓ (based on visual inspection only; trust accordingly) Tests: not run Approval: ✅ Comment: I did leave 2 comments on 2 separate lines in the diffs; feel free to read, respond, or even ignore as I don't think they're critical

Thanks @nutjob4life for your review!
